### PR TITLE
DHFPROD-5234: Compare values of each matched pair

### PR DESCRIPTION
**Pull request by AkshayBajajML**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by AkshayBajajML** [comment]
@jbelonoj reviewed and approved UI changes

File: marklogic-data-hub-central/ui/src/components/entities/matching/compare-values-modal/compare-values-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+                    return parseStructuredProperty(entityDefinitionsArray, structProperty, parentDefinitionName, immediateParentKey, parentKeysTempArray, allParentKeysTempArray, propertyPathUri, i, localParentKey);
+                  } else {
+                    let allParentKeysArray = [...allParentKeysTempArray];
+                    let updatedPropertyPath = getPropertyPath(parentKeysTempArray, structuredType.name, structProperty.name, propertyPath, i, property.name);
+                    let propertyValueInURI1 = propertyValueFromPath(updatedPropertyPath, property1);
+                    let propertyValueInURI2 = propertyValueFromPath(updatedPropertyPath, property2);
+                    let localPropertyPath = getPropertyPathForStructuredProperties(allParentKeysArray, structProperty.name);
+                    let matchedRow = propertyValueInURI1 && propertyValueInURI2 && (JSON.stringify(propertyValueInURI1) === JSON.stringify(propertyValueInURI2)) ? matchedProperties.includes(localPropertyPath) : false;
```
Comment (xnikhil08) [inline marklogic-data-hub-central/ui/src/components/entities/matching/compare-values-modal/compare-values-modal.tsx:176]: Can we test this without JSON.stringify equality comparison ? Just remove the condition "(JSON.stringify(propertyValueInURI1) === JSON.stringify(propertyValueInURI2))" and see if you get the proper results. 
The equality comparison might throw us off in few edge cases where the match type is not Exact.

**Review by xnikhil08 — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/entities/matching/compare-values-modal/compare-values-modal.tsx (hunk 1 above)
Comment (AkshayBajajML) [reply marklogic-data-hub-central/ui/src/components/entities/matching/compare-values-modal/compare-values-modal.tsx:176]: Yes, it returns same results after removing mentioned condition. I will update my PR and will push changes again.

**Review by xnikhil08 — APPROVED** [review]
Looks good.

**Review by Sanjeevani19 — APPROVED** [review]

**Pull request merged**